app/trip.py

```diff
@@ -0,0 +1,101 @@
+import datetime
+from app.shop import Shop
+from app.customer import Customer
+
+
+class Trip:
+
+    @staticmethod
+    def choosing_the_cheapest_shop(
+            customer: Customer,
+            shops: list[Shop],
+            fuel_price: float
+    ) -> Shop or None:
+        trip_prices = {}
+        for shop in shops:
+            specific_shop_trip_price = Trip.calculate_costs(
+                customer, shop, fuel_price
+            )
+            trip_prices[specific_shop_trip_price] = shop
+
+            print(
+                (f"{customer.name}'s trip to the {shop.name} "
+                 f"costs {specific_shop_trip_price}")
+            )
+        lowest_trip_cost = min(trip_prices)
+        if customer.money >= lowest_trip_cost:
+            return trip_prices[lowest_trip_cost]
+
+    @staticmethod
+    def shopping(shop: Shop, customer: Customer) -> str:
+        milk_amount, milk_price = (customer.product_cart["milk"],
+                                   shop.products["milk"])
+        bread_amount, bread_price = (customer.product_cart["bread"],
+                                     shop.products["bread"])
+        butter_amount, butter_price = (customer.product_cart["butter"],
+                                       shop.products["butter"])
+        total_cost = ((milk_amount * milk_price)
+                      + (bread_amount * bread_price)
+                      + (butter_amount * butter_price))

```

Comment:
try not to hardcode a products, what if product cart will change ? use loop for this